shared/reducers/map.js

```diff
@@ -43,6 +46,29 @@ const map = (state = initialState, action = {}) => {
     return update(state, {
       tabview_index: { $set: action.index }
     });
+  case types.setNavigator:

```

Comment:
Why you create at map-reducer? since, this is navigator-role.
What do you think about create new reducer such like 'navigator-reducer'.